Mage.Sets/src/mage/cards/t/TibaltWickedTormentor.java

```diff
@@ -0,0 +1,123 @@
+ */
+public final class TibaltWickedTormentor extends CardImpl {
+
+    private static final List<String> spellbook = Collections.unmodifiableList(Arrays.asList(
+            "Brimstone Vandal",
+            "Chained Brute",
+            "Charmbreaker Devils",
+            "Devil's Play",
+            "Festival Crasher",
+            "Forge Devil",
+            "Frenzied Devils",
+            "Havoc Jester",
+            "Hellrider",
+            "Hobblefiend",
+            "Pitchburn Devils",
+            "Sin Prodder",
+            "Spiteful Prankster",
+            "Tibalt's Rager",
+            "Torch Fiend"
+    ));
+
+    public TibaltWickedTormentor(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.PLANESWALKER}, "{3}{R}{R}");
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.TIBALT);
+
+        this.setStartingLoyalty(3);
+
+        // +1: Add {R}{R}. Draft a card from Tibalt, Wicked Tormentor’s spellbook, then exile it. Until end of turn, you may cast that card.
+        Ability ability = new LoyaltyAbility(new BasicManaEffect(Mana.RedMana(2)), 1);
+        ability.addEffect(new DraftFromSpellbookThenExileCastThisTurnEffect(spellbook));
+        this.addAbility(ability);
+
+        // +1: Tibalt, Wicked Tormentor deals 4 damage to target creature or planeswalker unless its controller has Tibalt deal 4 damage to them.
+        // If they do, you may discard a card. If you do, draw a card.
+        Ability ability1 = new LoyaltyAbility(new TibaltDamageEffect(), 1);
+        ability1.addTarget(new TargetCreatureOrPlaneswalker());
+        this.addAbility(ability1);
+
+        // -X: Create X 1/1 red Devil creature tokens with “When this creature dies, it deals 1 damage to any target.”
+        this.addAbility(new LoyaltyAbility(new CreateTokenEffect(new DevilToken(), GetXValue.instance)));
+    }
+
+    private TibaltWickedTormentor(final TibaltWickedTormentor card) {
+        super(card);
+    }
+
+    @Override
+    public TibaltWickedTormentor copy() {
+        return new TibaltWickedTormentor(this);
+    }
+}
+
+class TibaltDamageEffect extends OneShotEffect {
+
+    TibaltDamageEffect() {
+        super(Outcome.Neutral);
+        this.staticText = " {this} deals 4 damage to target creature or planeswalker unless its controller has Tibalt deal 4 damage to them." +
+                " If they do, you may discard a card. If you do, draw a card.";
+    }
+
+    private TibaltDamageEffect(final TibaltDamageEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public TibaltDamageEffect copy() {
+        return new TibaltDamageEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getFirstTarget());
+        Player player = game.getPlayer(source.getControllerId());
+        Player controller = game.getPlayer(permanent.getControllerId());
```

Comment:
1. Miss permanent null check
2. Looks like you wrongly use player and controller (must invert). Also check who does damage and make an answer.